### PR TITLE
Add get_response method as a synchronous alternative to converse

### PR DESCRIPTION
**Pull request by MatthewScholefield**
Documentation:
```python
def get_response(self, dialog, data=None):
    """
    Ask and return a response from the user for the given dialog

    Args:
        dialog (str): Dialog file to read to the user
        data (str): data used to render the dialog
    Returns:
        any: None or transcription user's reply
    """
    pass
```

Example usage:
```python
color = self.get_response('what.is.your.favorite.color')
if color:
    self.speak('My favorite color is also ' + color)
```

### CODE REVIEW

File: mycroft/client/speech/main.py [hunk 1]
```diff
@@ -56,6 +56,10 @@ def handle_utterance(event):
     ws.emit(Message('recognizer_loop:utterance', event))
 
 
+def handle_unknown():
+    ws.emit(Message('speech.recognition.unknown'))
```
Comment (forslund) [inline mycroft/client/speech/main.py]: @penrods suggested using recognizer_loop:x internally in the speech client and using mycroft.x.y on the interprocess message bus.

Comment (MatthewScholefield) [reply mycroft/client/speech/main.py]: OK, this has been changed to `recognizer_loop:speech.recognition.unknown` and `mycroft.speech.recognition.unknown`. Let me know if any other name sounds better.

File: mycroft/client/speech/listener.py [hunk 2]
```diff
@@ -152,6 +152,7 @@ def transcribe(self, audio):
                 text = "pair my device"  # phrase to start the pairing process
                 LOG.warning("Access Denied at mycroft.ai")
         except Exception as e:
+            self.emitter.emit('recognizer_loop:peech.recognition.unknown')
```
Comment (penrods) [inline mycroft/client/speech/listener.py]: Is "peech" intentional?

File: mycroft/client/speech/main.py [hunk 3]
```diff
@@ -56,6 +56,10 @@ def handle_utterance(event):
     ws.emit(Message('recognizer_loop:utterance', event))
 
 
+def handle_unknown():
+    ws.emit(Message('mycroft.mycspeech.recognition.unknown'))
```
Comment (penrods) [inline mycroft/client/speech/main.py]: Is "mycspeech" intentional?  Looks like a search/replace error.

Comment (MatthewScholefield) [reply mycroft/client/speech/main.py]: Lol, no idea how I failed at typing that much. Fixed now.

File: mycroft/skills/core.py [hunk 4]
```diff
@@ -320,13 +322,121 @@ def converse(self, utterances, lang="en-us"):
...
+
+        # TODO: Load with something like mycroft.dialog.get_all()
+        cancel_voc = 'text/' + self.lang + '/cancel.voc'
+        with open(resolve_resource_file(cancel_voc)) as f:
+            cancel_words = list(filter(bool, f.read().split('\n')))
+
+        def is_cancel(utterance):
+            return utterance in cancel_words
+
+        def on_valid_default(utterance):
```
Comment (MatthewScholefield) [inline mycroft/skills/core.py]: Sounds a bit funny since the original parameter isn't called `on_valid`.

**Pull request merged**